version: "3.8"
services:
  postgres:
    image: postgres:10.18-alpine
    volumes:
      - "$PWD/docker-pgsql-entrypoint:/docker-entrypoint-initdb.d"
      - "$PWD/docker-pgsql-volume:/var/lib/postgresql/data" 
    ports:
      - 5432:5432
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=testing_password
      - TESTING_POSTGRES_USER=testing_user
      - TESTING_POSTGRES_DB=testing_metadb
    deploy:
      restart_policy:
        condition: on-failure
        max_attempts: 3
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 15s
      timeout: 5000s
      retries: 50
  testing:
    build:
      context: ./
    container_name: metadb
    image: terraform-aws-infrastructure-live-ci-unit-testing:latest
    volumes:
      - "$PWD/../../:/src"
    environment:
      - PGUSER=testing_user
      - PGPASSWORD=testing_password
      - PGDATABASE=testing_metadb
      - PGHOST=postgres
      - PGPORT=5432
      - ADDITIONAL_PATH=/src/node_modules/.bin
    command: ["/bin/bash"]
    entrypoint: ["/bin/bash", "/src/tests/unit/entrypoint.sh"]
    depends_on: 
      - postgres
    